### ISSUE TITLE
User Prompted to Retreat Planes to current space during battle.

### ISSUE THREAD

**Issue by tvleavitt** [triplea-game/triplea#7888]
### How can the problem be recreated?

I suspect running through the tutorial would be enough. I saw this several times, actually. One time I think  I recall with the Italians invading Great Britain (not Scotland). 

![TripleA-tutorial-Italians-attacking-Egypt-retreat-planes-to-Egypt-asked](https://user-images.githubusercontent.com/4445057/95730954-c7ce8d00-0c33-11eb-98d9-944ed066cd89.png)

### Do you have any ideas for an expected fix?

No prompts to retreat planes into the space they're currently attacking.

### Attach a Save Game

I was focused on playing the game and spaced out 

on the idea of creating a save game when I did the screen capture.

### If playing a prerelease, which version are you using?
2.3.1953

Game Version: 1.0.0 ... which is wonky, since I know this has been updated?

### If playing a prerelease, does this happen on the latest release?

Haven't tested this. Apologies.

- [] yes
- [] no

**Comment by Cernelius** [comment]
You are right that the behaviour is wrong, but I believe you are asking for a wrong solution to it.

I remember that the tutorial is supposed to be v2 (I don't know if this is clarified, and I've not checked it).

The solution that you want is how it should work if the tutoral would be v1 (which I believe it isn't) (I point out that here I'm not considering what should happen if you enable air retreating in v1).

The tutorial is v2 rules, under which you should be able to retreat all air units, but you should decide where to retreat and make the retreat movement right at that point (resolving eventualy fly-over fire at the end of the Conduct Combat phase, after all battles are done), then the combat will continue with only land units involved (that can never retreat, as long as one or more of those you had in the battle at the start of it were offloaded). Practically, instead of giving the "Retrate to Egypt" option, the program should make you able to retreat every air unit wherever it can land (the decision can be made singularly, per air unit) (during Non Combat Move, you can never move air units that you moved during Combat Move). The only item that I'm not sure about is what should happen in the case you have air units starting their turn into a hostile sea zone and launching to engage in battle, without moving, but I tend to think that they cannot move during Non Combat Move too (I would need an official clarification here).

According to your description, how the tutorial behaves is v2 LHTR, and later, rules, which I assume is a wrong behaviour, as long as the tutorial is not v2 LHTR, nor later, rules.

@panther2 Can you check that what I said is all right.

As long as what I said is correct, the first post of this issue should be reworded or the issue closed.

**Comment by trevan** [comment]
This occurs when you do an amphibious assault that includes planes.  If "WW2V2", "Attacker Retreat Planes", or "Partial Amphibious Retreat" are set, then the code will allow you to retreat the planes after each round.  Currently, all plane retreats always go to the territory that the battle is in (you can try this with a normal retreat that has planes).  After Combat Move, you'll then be able to move the planes to a different territory.

**Comment by Cernelius** [comment]
> This occurs when you do an amphibious assault that includes planes. If "WW2V2", "Attacker Retreat Planes", or "Partial Amphibious Retreat" are set, then the code will allow you to retreat the planes after each round. Currently, all plane retreats always go to the territory that the battle is in (you can try this with a normal retreat that has planes). After Combat Move, you'll then be able to move the planes to a different territory.

This is how it works in v2 LHTR and later. Anyway, it is very much of a minor matter, in my opinion.

I assume that "WW2V2" sets "Attacker Retreat Planes" true, but doesn't influence "Partial Amphibious Retreat".

**Comment by tvleavitt** [comment]
From the naive end user's perspective (regardless of rules), asking to have planes retreat to the territory they are attacking, which they aren't supposed to be able to land in, and which might well be occupied by an enemy force is nonsensical without explaining that you'll be able to move them during the non-combat phase (you might assume that the user will connect the dots here, but I'm a fairly experience player and did not). Especially when sea battles and other retreats always prompt you to retreat to a different territory.

**Comment by Cernelius** [comment]
> From the naive end user's perspective (regardless of rules), asking to have planes retreat to the territory they are attacking, which they aren't supposed to be able to land in, and which might well be occupied by an enemy force is nonsensical without explaining that you'll be able to move them during the non-combat phase (you might assume that the user will connect the dots here, but I'm a fairly experienced player and did not). Especially when sea battles and other retreats always prompt you to retreat to a different territory.

In theory, those pieces are not in Egypt, during the battle. They are on the battleboard. So, you would remove all your aircraft pieces from the battleboard and place them back in Egypt, then go on with the battle with what you have left on the battleboard.

**Comment by Cernelius** [comment]
Maybe, instead of "Retreat to Egypt", it can be "Return all aircraft to Egypt" or "Remove all aircraft from battle". Not sure what's better.

**Comment by trevan** [comment]
> Maybe, instead of "Retreat to Egypt", it can be "Return all aircraft to Egypt" or "Remove all aircraft from battle". Not sure what's better.

What about just "Retreat"?

**Comment by tvleavitt** [comment]
This makes sense: "Remove all aircraft from battle." Then, during the non-combat move phase, they can fly elsewhere as normal, and it works just like any other post-battle situation.

**Comment by tvleavitt** [comment]
To me, "Retreat" implies sending units to another territory, which is not what we're talking about.

Is this a "glossary" issue?

**Comment by trevan** [comment]
> I assume that "WW2V2" sets "Attacker Retreat Planes" true, but doesn't influence "Partial Amphibious Retreat".

I don't believe "WWV2" sets anything.  The check for this code checks if any of those three properties are set.

**Comment by beelee1** [comment]
yea threw me off a bit first time I saw it. I know how it works now so don't really think about it. Could also use "Disengage" instead of Remove. Doesn't really matter much. Just a phrasing thing.

**Comment by trevan** [comment]
> To me, "Retreat" implies sending units to another territory, which is not what we're talking about.
> 
> Is this a "glossary" issue?

The change for this text will probably have to wait till 3.0, then.  I'm not sure if it can be done in a non-compatible manner, unfortunately.  Can you update the title so that it mentions that this issue is to change the prompt?

**Comment by DanVanAtta** [comment]
What if we updated the retreat button to not mention the territory?

**Comment by trevan** [comment]
> What if we updated the retreat button to not mention the territory?

It currently will display the retreat territory if there is only one option available.  If there are more than two, it will just say "Retreat" and then when you click that button, it will open a dialog to select the retreat territory.

The problem with the non-compatibility is that the request for the retreat goes through a RemoteActionCode and the only information sent is the list of territories and whether it was a submerge request.  So there is no information on if it was a partial air retreat to be able to give a better string.

**Comment by tvleavitt** [comment]
> > To me, "Retreat" implies sending units to another territory, which is not what we're talking about.
> > Is this a "glossary" issue?
> 
> The change for this text will probably have to wait till 3.0, then. I'm not sure if it can be done in a non-compatible manner, unfortunately. Can you update the title so that it mentions that this issue is to change the prompt?

I updated the title.

**Comment by tvleavitt** [comment]
> > What if we updated the retreat button to not mention the territory?
> 
> It currently will display the retreat territory if there is only one option available. If there are more than two, it will just say "Retreat" and then when you click that button, it will open a dialog to select the retreat territory.
> 
> The problem with the non-compatibility is that the request for the retreat goes through a RemoteActionCode and the only information sent is the list of territories and whether it was a submerge request. So there is no information on if it was a partial air retreat to be able to give a better string.

My interpretation is that this is "generic" retreat code that isn't "plane aware", and that for whatever reason, it interprets a plane retreat as only being able to retreat to the current territory, thus it displays the current territory name. Am I correct?

**Comment by trevan** [comment]
> > > What if we updated the retreat button to not mention the territory?
> > 
> > 
> > It currently will display the retreat territory if there is only one option available. If there are more than two, it will just say "Retreat" and then when you click that button, it will open a dialog to select the retreat territory.
> > The problem with the non-compatibility is that the request for the retreat goes through a RemoteActionCode and the only information sent is the list of territories and whether it was a submerge request. So there is no information on if it was a partial air retreat to be able to give a better string.
> 
> My interpretation is that this is "generic" retreat code that isn't "plane aware", and that for whatever reason, it interprets a plane retreat as only being able to retreat to the current territory, thus it displays the current territory name. Am I correct?

There's two parts of the retreat code.  There's one that is "partial-plane amphibious retreat" aware and it recognizes that there are only planes that can retreat.  Since it knows that the planes are retreating and not the other units, it knows that the planes will "retreat" (or be removed or whatever term we want to use) to the battle location so that they can be later moved.  But it can only send the retreat locations and whether it was a submerge to the "remote user".

At the remote user, it only knows the battle location, the retreat locations, and whether there is a submerge.  And it uses this information to build the dialog that is shown to the user.  If more than one retreat option is sent or it isn't a submerge, it shows the retreat dialog.  If the submerge option is sent, it shows the submerge option.

If we could assume that any retreat request that 1) isn't a submerge request and 2) only has one retreat option and 3) the retreat option is equal to the battle territory are always a plane-only retreat, then we could change the text to say something like "Remove all aircraft".  But if aren't sure that that is always plane-only retreats, then the text would need to be a little more generic.  That's why I asked about just saying "Retreat".

**Comment by trevan** [comment]
I went through all of the retreat logic and there's only two scenarios when there's only one retreat option and it is equal to the battle site.
* When all the units retreating are air based
* When the property "Retreating Units Remain In Place" is true

So, does any one have a suggestion on what text to put on the button that will be less confusing but handle those two scenarios?

**Comment by Cernelius** [comment]
>"Planes" instead of "Air Units" ?

While in World War 2 almost all (but not all, as balloon dirigibles were used too (and a few balloon bombs)) (I'm not counting barrage balloons as air units, as I tend to see them as anti-air surface units) air units were "planes" (short for airplane (American English) or aeroplane (British English)), this is not necessarily generally the case, unless TripleA is meant to capture WW2 only (which it may be argued it is, as there are many instances in which it appears it does). So, using "planes" may be right or wrong depending whether it is wanted or not wanted for TripleA to be meaningful for WW2 only.

However, one can also argue that the current TripleA air units are substantially only capturing the behaviour of heavier-than-air units, or even less than that (for example, it can be easily argued that TripleA air units are not representing helicopters, ballistic missiles, birds and others), so they should rather be called airplanes or aeroplanes. In this case, one could say that the current TripleA air units are indeed only aeroplanes, and any other kind of air unit would be simply currently not supported.
